pkg/cmd/cli/cmd/env.go

```diff
@@ -324,24 +333,28 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 
 	failed := false
 	for i, info := range infos {
-		newData, err := json.Marshal(objects[i])
-		if err != nil {
-			return err
-		}
-		patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData[i], newData, objects[i])
-		if err != nil {
-			return err
-		}
-		obj, err := resource.NewHelper(info.Client, info.Mapping).Patch(info.Namespace, info.Name, kapi.StrategicMergePatchType, patchBytes)
-		if err != nil {
-			handlePodUpdateError(cmd.Out(), err, "environment variables")
-			failed = true
-			continue
+		outputMsg := "updated"
+		if !updated[i] {
+			outputMsg = "nothing to update"
```

Comment:
Instead of doing this and nesting all the code, do this:
```go
if !updated[i] {
   cmdutil.PrintSuccess(mapper, shortOutput, out, info.Mapping.Resource, info.Name, "didn't need to update")
   continue
}
// rest of the code...
```